lib/libxdp/tests/test_link_dispatch.c

```diff
@@ -0,0 +1,111 @@
+ /* SPDX-License-Identifier: GPL-2.0 */
+  
+#define _GNU_SOURCE
+
+#include <errno.h>
+#include <linux/err.h>
+#include <net/if.h>
+#include <pthread.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/resource.h>
+#include <unistd.h>
+
+#include "test_utils.h"
+
+#include <xdp/libxdp.h>
+#include <bpf/libbpf.h>
+
+
+static void usage(char *progname)
+{
+	fprintf(stderr, "Usage: %s <ifname>\n", progname);
+	exit(EXIT_FAILURE);
+}
+
+static int check_link_detach(int ifindex) {
```

Comment:
The indentation is off in this and the `main()` function below. We use a single tab character per indentation level everywhere.